test/RpcEvmCompatibility.js

```diff
@@ -54,14 +54,46 @@ describe('EVM RPC Compatibility Test', () => {
         const newValue = 100;
         const receipt = await rpcCompatibilityContract.updateValue(newValue).then((tx) => tx.wait());
         const blockNumber = hexValue(receipt.blockNumber);
+        const logs = [];
 
-        const filter = {
+        let filter = {
             fromBlock: blockNumber,
             toBlock: blockNumber,
         };
-        const logs = await provider.send('eth_getLogs', [filter]);
-        expect(logs).to.be.an('array');
-        expect(logs.length).to.be.greaterThan(0);
+        let log = await provider.send('eth_getLogs', [filter]);
+        logs.push(log);
+        filter = {
+            fromBlock: blockNumber,
+            toBlock: 'latest',
+        };
+        log = await provider.send('eth_getLogs', [filter]);
+        logs.push(log);
+        filter = {
+            fromBlock: blockNumber,
+            toBlock: 'pending',
+        };
+        log = await provider.send('eth_getLogs', [filter]);
+        logs.push(log);
+
+        if (network.name.toLowerCase() === 'ethereum') {
+            filter = {
+                fromBlock: blockNumber,
+                toBlock: 'safe',
```

Comment:
can you create a dedicated test for safe/finalized tag to distinguish between supporting the rpc method and supporting the tag?

Reply:
We care about it for eth_getLogs, and eth_getBlockByNumber since as a cross-chain provider we're checking for finalized tx events at the gateway.

`latest` is needed for all the RPC methods. And `safe`/`finalized` can be checked within a dedicated test since it's only available on a subset of chains.